File: sa p/MultiHotelManagement/src/main/java/com/MHM/MultiHotelManagement/service/FacilityService.java
package com.MHM.MultiHotelManagement.service;

import com.MHM.MultiHotelManagement.dto.request.FacilityRequestDTO;
import com.MHM.MultiHotelManagement.dto.response.FacilityResponseDTO;

import java.util.List;

public interface FacilityService {

    // একটা facility তৈরি করো
    FacilityResponseDTO create(FacilityRequestDTO dto);

    // একসাথে অনেকগুলো facility তৈরি করো
    List<FacilityResponseDTO> createBulk(
            Long hotelId,
            List<FacilityRequestDTO> dtoList
    );

    // সব facility
    List<FacilityResponseDTO> getAll();

    // ID দিয়ে খোঁজো
    FacilityResponseDTO getById(Long id);

    // Hotel ID দিয়ে সব facility খোঁজো
    List<FacilityResponseDTO> getByHotelId(Long hotelId);

    // Update করো
    FacilityResponseDTO update(Long id, FacilityRequestDTO dto);

    // একটা delete করো
    void delete(Long id);

    // Hotel এর সব facility delete করো
    void deleteAllByHotelId(Long hotelId);
}

File: sa p/MultiHotelManagementAngular/src/app/services/facility.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Facility, FacilityRequest } from '../models/facility.model';
import { environment } from '../../environments/environments';

@Injectable({ providedIn: 'root' })
export class FacilityService {
  private API_URL = environment.apiUrl + 'facilities';

  constructor(private http: HttpClient) {}

  getAll() {
    return this.http.get<Facility[]>(this.API_URL);
  }

  getById(id: number) {
    return this.http.get<Facility>(`${this.API_URL}/${id}`);
  }

  getByHotel(hotelId: number) {
    return this.http.get<Facility[]>(`${this.API_URL}/hotel/${hotelId}`);
  }

  create(data: FacilityRequest) {
    return this.http.post<Facility>(this.API_URL, data);
  }

  createBulk(hotelId: number, data: FacilityRequest[]) {
    return this.http.post<Facility[]>(`${this.API_URL}/bulk/${hotelId}`, data);
  }

  update(id: number, data: FacilityRequest) {
    return this.http.put<Facility>(`${this.API_URL}/${id}`, data);
  }

  delete(id: number) {
    return this.http.delete<void>(`${this.API_URL}/${id}`);
  }

  deleteByHotel(hotelId: number) {
    return this.http.delete<void>(`${this.API_URL}/hotel/${hotelId}`);
  }
}
